src/main/scala/zio/prelude/Associative.scala

```diff
@@ -31,6 +33,15 @@ import zio.{ Chunk, NonEmptyChunk }
  */
 trait Associative[A] {
   def combine(l: => A, r: => A): A
+
+  final def repeatOption(a: A, n: Int): Option[A] = {
```

Comment:
Should the semantics here just be that if you give us a number less than or equal to one we return the original value unchanged? That way we wouldn't need the option.

Reply:
Maybe it would be best to start with the behavior of `repeat`. I would like the following property:

Assuming `n` and `m` are natural numbers,
```
repeat(n) <> repeat(m)  ===  repeat(n + m)
```

I don't think that property would hold, if `repeat(a, 0)` === `a`.

And `repeatOption` should behave analogously.

What do you think about it? :thinking:

Reply:
We do not have support for positive numbers. If we have, we could:
`repeat(Positive(n)) <> repeat(Positive(m))  ===  repeat(Positive(n) + Positive(m))`
Currently we could have 
`repeat(max(1,n)) <> repeat(max(1,m))  ===  repeat(max(1,n) + max(1,m))`
to keep semantics that @adamgfraser proposed. Also since that was my idea on the start. So maybe this is more intuitive even though law is a bit more complex.

I do not mind sorting this out in next PR :)

Reply:
I don't have strong feelings about this either way. But if we take into consideration not Associative, not Identity, but `Inverse` specifically, we wouldn't need to restrict ourselves to Naturals. We could generalize this even further.

For any Integer, this holds (or could hold):
```
repeat(n) <> repeat(m)  ===  repeat(n + m)
```
where `repeat(a, 0)` === `id` and `repeat(a, -2)` === `inverse(inverse(id, a), a)`, etc...

If we make a step back, what we're trying to do here with `repeat` is multiplication.
And only `Inverse` elements can be "multiplied" by a negative factor.
`Identity` elements cannot do that, but can at least be multiplied by 0.
`Associative` elements cannot do even that and have to be multiplied only by > 0.

If we wanted to keep with such semantics, we would need to have `repeatOption` on both `Associative` and `Identity` and `repeat` only on `Inverse`.

Do you think that would make sense?

Reply:
I like the idea, but I would love to see this in separate PR as this is going into the design not ment for beginner PR.

Reply:
I guess I am coming at it from a little bit of a different perspective. I am thinking about methods like `repeat` operators on ZIO where we have for example:

```scala
  final def repeatN(n: Int): ZIO[R, E, A] =
    self.flatMap(a => if (n <= 0) ZIO.succeedNow(a) else ZIO.yieldNow *> repeatN(n - 1))
```

So this is basically saying if you ask us to repeat the effect zero or fewer times we are just going to ignore you and give you back the original value you gave us. I also see this as a case where calling this operator is almost a case of caller error, like doing `foreachPar` with a negative number, and something that if we were in the effectual world would warrant a `Die` cause. So returning a pretty logical value back seems like a very benign response to that. I also think having the `Option` in the return type creates significant user friction in working with this operator if they want to actually do something else with the value.

If the user really cares about guarding for the negative case they can always check the value before calling `repeat` and handle it however they wish.

Reply:
OK, I think for `repeat` (or should we rename it to `repeatN`?) we should implement it as in ZIO, as @adamgfraser says.
We'll deal with `multiply` in another ticket.
Sounds good?

Reply:
Sounds good!

Reply:
Here's the new ticket https://github.com/zio/zio-prelude/issues/417